extensions/wikia/VideoHandlers/js/handlers/Ooyala.js

```diff
@@ -79,55 +81,97 @@ define('wikia.videohandler.ooyala', [
 			});*/
 		}
 
+		function loadJs(resource) {
+			return loader({
+				type: loader.JS,
+				resources: resource
+			}).fail(loadFail);
+		}
+
+		function initRegularPlayer() {
+			log('Create Ooyala player', log.levels.info, logGroup);
+
+			win.OO.Player.create(containerId, params.videoId, createParams);
+		}
+
 		createParams.onCreate = onCreate;
 
 		if (adContext && adContext.getContext().opts.showAds) {
 			if (!dartVideoHelper) {
 				throw 'ext.wikia.adEngine.dartVideoHelper is not defined and it should as we need to display ads';
 			}
 
-			tagUrl = dartVideoHelper.getUrl();
-
 			createParams.vast = {
-				tagUrl: tagUrl
+				tagUrl: dartVideoHelper.getUrl()
 			};
 		}
 
 		// log any errors from failed script loading (VID-976)
-		function loadFail( data ) {
+		function loadFail(data) {
 			var message = data.error + ':';
 
-			$.each( data.resources, function() {
+			$.each(data.resources, function () {
 				message += ' ' + this;
 			});
 
-			log( message, log.levels.error, 'VideoBootstrap' );
+			log(message, log.levels.error, logGroup);
+		}
+
+		function initRecoveredPlayer() {
+			log('Create recovered Ooyala player', log.levels.info, logGroup);
+
+			win.googleImaSdkFailedCbList = {
+				originalCbList: [],
+				unshift: function (cb) {
```

Comment:
there is a difference between `unshift` and `push`. Why the property is called `unshift` but `push` is called inside?